src/main/java/com/google/sps/servlets/BusLocations.java

```diff
@@ -0,0 +1,98 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.sps.data.CoordinateCalculator;
+import com.google.sps.data.EntityType;
+import com.google.sps.data.Office;
+import com.google.sps.data.OfficeManager;
+import com.google.sps.data.QueryHelper;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that handles adding and retreiving listings & locations */
+@WebServlet("/busLocations")
+public class BusLocations extends HttpServlet {
+
+  private final int numStops = 10;
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    String office = request.getParameter("office");
+    Office selectedOffice = OfficeManager.offices.get(office);
```

Comment:
what happens if OfficeManager can not find office?

Reply:
So on the front end, the user selects an office from a pre-made list. Because of that the only way the office manager wouldn't be able to find the office is if the user messes with the URL parameters directly or if someone updates the front end pre made list and forgets to update the backend. 

If that happens, I believe the server will throw a 500 error and MVN will show the stacktrace to this line. Do you think I should have it throw a custom error instead?